apps/sel4test-tests/src/tests/faults.c

```diff
@@ -1069,6 +1069,353 @@ static int test_timeout_fault_nested_servers(env_t env)
 /* this test is disabled for the same reason as TIMEOUTFAULT0002 */
 DEFINE_TEST(TIMEOUTFAULT0003, "Nested timeout fault", test_timeout_fault_nested_servers, config_set(CONFIG_KERNEL_MCS))
 
+/* Test of invalid SC and no SC faults */
```

Comment:
Did you consider adding test cases for where each of these scenarios *doesn't* have a fault handler? we want to make sure the thread still becomes inactive and the kernel doesn't die for some reason.

Reply:
I don't know of a way to effectively check that a thread isn't still running, but can certainly check that the kernel doesn't explode (or fail any asserts).